x-pack/filebeat/input/awss3/input.go

```diff
@@ -410,5 +413,35 @@ func pollSqsWaitingMetric(ctx context.Context, receiver *sqsReader) {
 	}
 }
 
+func pollSqsUtilizationMetric(ctx context.Context, receiver *sqsReader, metricReporterChan chan int64) {
+	defer close(metricReporterChan)
```

Comment:
Is there any chance that the `sqsReceiver.Receive()` continues to run after this channel is closed? That will result in a panic if the goroutine tries to write to the closed channel. Normally I tie the life of the channel to the life of the writing goroutines. Readers can automatically safely stop after then channel is closed and drained.

Reply:
I don't think so, but I moved the close to _I think_ better fit according to the writing goroutines.